scripts/background/AudioManager.js

```diff
@@ -164,6 +169,54 @@ function AudioManager(addEventListener, isTownTune) {
 
 	addEventListener("volume", newVol => {
 		audio.volume = newVol;
+		setVolume = newVol;
+	});
+
+	// If a tab starts or stops playing audio
+	addEventListener("tabAudio", (audible, tabAudio, reduceValue) => {
+		if (audible != null) {
+			// Handles all cases except for an options switch.
+			if (tabAudio == 'pause') {
+				if (audible) {
+					if (!audio.paused) {
+						audio.pause();
+						tabAudioPaused = true;
+					}
+				} else {
+					if (audio.paused) {
+						audio.play();
+						tabAudioPaused = false;
+						// Get the badge icon updated.
+						window.notify("unpause");
+					}
+				}
+			}
+
+			if (tabAudio == 'reduce') {
+				if (audible) {
+					let newVolume = setVolume * (1 - reduceValue / 100);
+					if (newVolume < 0) newVolume = 0;
+					audio.volume = newVolume;
+					reducedVolume = true;
+				} else {
+					audio.volume = setVolume;
+					reducedVolume = false;
+				}
+			}
+		} else {
+			// Handles when the options are switched. Disables the previous option and enables the new one.
+
+			if (audio.paused && tabAudio != 'pause') {
+				audio.play();
+				tabAudioPaused = false;
+				window.notify("unpause");
+			}
+			if (reducedVolume && tabAudio != 'reduce') {
+				reducedVolume = false;
+				audio.volume = setVolume;
+			}
+			window.notify("tabAudio", [true, tabAudio, reduceValue]);
```

Comment:
Current logic introduces following bug:
If music **from the extension** is playing (and no other tabs are playing music), when a 'Tab Audio Behaviour' option is switched to 'reduce' /'pause', extension music is reduced/paused. 

That is, switching to 'pause' will effectively pause the extension even though there are no other sounds.